### PR TITLE
chore(selector): add CFNSelector

### PR DESCRIPTION
**Pull request by efekarakus**
This PR  should be reviewed after #4540.
The commit that has the change is d6b8265f91a9ced608ee2fd06075ccca85c70292.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/term/selector/cfn.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		} `yaml:"Resources"`
+	}{}
+	if err := yaml.Unmarshal([]byte(body), &tpl); err != nil {
+		return nil, fmt.Errorf("unmarshal CloudFormation template: %v", err)
+	}
+
+	// Prompt for a selection.
+	var options []prompt.Option
+	for name, resource := range tpl.Resources {
+		if resource.Type == "AWS::Lambda::Function" || strings.HasPrefix(resource.Type, "Custom::") {
```
Comment (iamhopaul123) [inline internal/pkg/term/selector/cfn.go:42]: Maybe it makes more sense to filter out from a specific list but I guess it's ok to be more restrictive at first.

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/term/selector/cfn.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/term/selector/cfn.go:42]: I kept debating if this method should take as input something like:
```go
func Resources(body string, filterFn ...func(template.CFNResource) bool) ([]template.CFNResource, error)
```
Happy to change it to that boss, if you prefer it to! It's definitely more flexible 👍 at the end I just kept it a bit simpler for now

Comment (iamhopaul123) [reply internal/pkg/term/selector/cfn.go:42]: Yeah i think we can keep things simple for now

**Review by paragbhingre — APPROVED** [review]
It looks good to me. Putting DNM just in case you decide to change the resource function definition But for me, it looks simple and easy to understand.

**Pull request merged**